spec/01-unit/03-prefix_handler_spec.lua

```diff
@@ -154,6 +154,14 @@ describe("NGINX conf compiler", function()
       assert.matches("worker_connections %d+;", nginx_conf)
       assert.matches("multi_accept on;", nginx_conf)
     end)
+    it("converts dns_resolver to string", function()
+      local nginx_conf = prefix_handler.compile_nginx_conf({
+        dns_resolver = { "8.8.8.8", "8.8.4.4" }
+      }, [[
+        "resolver ${{DNS_RESOLVER}} ipv6=off;"
```

Comment:
Hmmm at the same time, it is true we do not need it anymore since all of our DNS resolution is currently taken care of on the Lua-land... 

> I mean that if user wants the Nginx resolver use the same resolvers as Kong DNS resolve

I feel like this is a completely valid point, and to sum up the concerns, I would say they are: 

1. some users may indeed want to avoid duplicating their DNS nameservers in both the Kong config and a custom Nginx template, if they need to. 
2. at the same time, it feels like we do not need Nginx to create a resolver if Kong never uses it... seems wasteful
3. (on the long term) I am a bit concerned by the fact that we wrap each cosocket `:connect()` call with our own Lua-land resolver. While it is nice to be able to use the same DNS resolver everywhere and feed it with better features than the Nginx one, I am curious about the performance impact of that, and wonder if we should not instead invest time and energy in the Nginx resolver itself (a long-time thought of mine)
4. this change makes our current configuration types compatible with legacy Nginx templates, such as the one in use by the user who raised this issue

Because of 2. and 4. (probably most of the use-cases), I think it's fine to merge this patch as-is, without adding it back to the template are I suggested to maybe do. 

Do you agree with this reasoning?